x-pack/plugin/sql/qa/mixed-node/src/test/java/org/elasticsearch/xpack/sql/qa/mixed_node/SqlCompatIT.java

```diff
@@ -197,4 +198,36 @@ private void assertCursorNotCompatibleAcrossVersions(Version version1, RestClien
         }
     }
 
+    public void testFormatTxtQueryWithCursorOnNewNode() throws IOException {
+        testFormatTxtQueryWithCursor(newNodesClient);
+    }
+
+    public void testFormatTxtQueryWithCursorOnOldNode() throws IOException {
+        testFormatTxtQueryWithCursor(oldNodesClient);
+    }
+
+    public void testFormatTxtQueryWithCursor(RestClient client) throws IOException {
+        // see https://github.com/elastic/elasticsearch/issues/83581
+        index("{\"foo\":1}", "{\"foo\":2}");
+
+        Request query = new Request("POST", "_sql");
+        XContentBuilder json = XContentFactory.jsonBuilder()
+            .startObject()
+            .field("query", "select foo from test order by foo")
```

Comment:
Are there any tests covering `GROUP BY` and pagination with txt format? If not, please add one.

Reply:
the nature of the query doesn't really matter for this test but I'll make this explicit with a `randomFrom(...)`